### DISCUSSION TITLE
Dashboard list panel should store folder selection as folderUid in panel model

### DISCUSSION THREAD

**Discussion by pdcleyn** [grafana/grafana#37016] [Dashboards, Panels & Visualizations]
**What happened**:
I’m trying to create a dashboard only consisting out of several dashboard list panels. Each of these panels should show dashboards from a specific folder.

All our dashboards and folders are created using file based provisioning (yaml files for folders and json files for dashboards). Everything lands in the right spot.

Thing is that, when creating the dashboard list, the selected folder translates to a folderID in json, to identify the folder to be used in a dashboard list panel. However, during provisioning, I can only define the folderUID, which cannot be used in the dashboard list to select the folder, it seems.

This basically means that when I use provisioning on an empty Grafana, all dashboards and folders are loaded properly, but the dashboard using the dashboard list, does not point to the right folder.
**What you expected to happen**:
I would expect I can either specify the folderID in the provisioning files or be able to use the folderUID in the dashboard list panel to select a dashboard folder.
**How to reproduce it (as minimally and precisely as possible)**:

* provision a folder with a dashboard
* create a 2nd dashboard, containing a dashboard list panel, pointing to the above created folder
* export the dashboard as a json file
* remove all dashboards again
* reintroduce, using provisioning, the initial dashboard and folder, together with the 2nd dashboard with the dashboard panel
* dashboard list will be empty 

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.4.4
- Data source type & version: n.a.
- OS Grafana is installed on: CentOS 6.9
- User OS & Browser: any
- Grafana plugins: n.a.
- Others:

**Comment by marefr** [comment]
There's some work getting this fixed. Contributions are welcome. 
- Current [FolderPicker component ](https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/dashlist/editor.html#L26)  only supports `folderId` currently. 
- We probably need a new react component to replace existing folder picker
- Support for searching by `folderUid` needs to be added to [search backend](https://github.com/grafana/grafana/blob/master/pkg/api/search.go) 
- Dashboard list panel needs to handle legacy `folderId` in panel model for backward compatibility and new `folderUid` in panel model.

**Comment by naturalbeau** [comment]
@marefr Is there any update on this issue? I also have the same issue and looks there is no way to work around it.

**Comment by chan-vince** [comment]
I have moved away from using provisioning because of this issue - instead I'm using a python application that uses the API to configure folders and dashboards after deployment, which significantly increases complexity.

Would be great to have this looked at because the provisioning system is pretty great otherwise! Thanks 😀

**Reply by alefteris** [reply]
Until this get fixed, a workaround is to set dashboard tags and filter by those in the dashboard list panels instead of filtering by folders. Filtering by tags doesn't have an issue with provisioned dashboards.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
